Classes/KIFUITestActor.m

```diff
@@ -471,21 +471,27 @@ - (void)enterTextIntoCurrentFirstResponder:(NSString *)text fallbackView:(UIView
     {
         if (![KIFTypist enterCharacter:characterString]) {
             // Attempt to cheat if we couldn't find the character
-            UIView * fallback = fallbackView;
-            if (!fallback) {
-                UIResponder *firstResponder = [[[UIApplication sharedApplication] keyWindow] firstResponder];
+            NSMutableArray<UIView *> *fallbackViews = [NSMutableArray array];
 
-                if ([firstResponder isKindOfClass:[UIView class]]) {
-                    fallback = (UIView *)firstResponder;
+            if (fallbackView) {
+                [fallbackViews addObject:fallbackView];
+            } else {
+                for (UIResponder *firstResponder in [[UIApplication sharedApplication] firstResponders]) {
+                    if ([firstResponder isKindOfClass:[UIView class]]) {
+                        [fallbackViews addObject:(UIView *)firstResponder];
+                    }
                 }
             }
 
-            if ([fallback isKindOfClass:[UITextField class]] || [fallback isKindOfClass:[UITextView class]] || [fallback isKindOfClass:[UISearchBar class]]) {
-                NSLog(@"KIF: Unable to find keyboard key for %@. Inserting manually.", characterString);
-                [(UITextField *)fallback setText:[[(UITextField *)fallback text] stringByAppendingString:characterString]];
-            } else {
-                [self failWithError:[NSError KIFErrorWithFormat:@"Failed to find key for character \"%@\"", characterString] stopTest:YES];
+            for (UIView *fallback in fallbackViews) {
+                if ([fallback isKindOfClass:[UITextField class]] || [fallback isKindOfClass:[UITextView class]] || [fallback isKindOfClass:[UISearchBar class]]) {
+                    NSLog(@"KIF: Unable to find keyboard key for %@. Inserting manually.", characterString);
+                    [(UITextField *)fallback setText:[[(UITextField *)fallback text] stringByAppendingString:characterString]];
```

Comment:
I wonder if we should print a warning if there is more than one match, to make the surface the somewhat arbitrary decision being made here.